Code/GraphMol/SmilesParse/SmilesParseOps.cpp

```diff
@@ -489,6 +485,12 @@ void CloseMolRings(RWMol *mol, bool toleratePartials) {
             swapBondDirIfNeeded(bond2, bond1);
             delete bond1;
           }
+
```

Comment:
This change doesn't make sense to me. It isn't connected to the memory leak found in the fuzzing (you fix that elsewhere) and leads to valgrind errors galore. Here's one of them:
```
==66850== Invalid read of size 4
==66850==    at 0x4A94320: RDKit::Bond::getIdx() const (Bond.h:164)
==66850==    by 0x50D98F3: RDKit::ROMol::clearBondBookmark(int, RDKit::Bond const*) (ROMol.cpp:264)
==66850==    by 0x4D560E9: SmilesParseOps::CloseMolRings(RDKit::RWMol*, bool) (SmilesParseOps.cpp:491)
==66850==    by 0x4D378E0: RDKit::toMol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int (*)(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<RDKit::RWMol*, std::allocator<RDKit::RWMol*> >&), std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (SmilesParse.cpp:218)
==66850==    by 0x4D38811: RDKit::SmilesToMol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, RDKit::SmilesParserParams const&) (SmilesParse.cpp:363)
==66850==    by 0x1C1ED9: RDKit::SmilesToMol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, bool, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >*) (SmilesParse.h:79)
==66850==    by 0x13D4B2: testPass() (test.cpp:119)
==66850==    by 0x1BDA96: main (test.cpp:4321)
==66850==  Address 0x637e610 is 48 bytes inside a block of size 80 free'd
==66850==    at 0x483CFBF: operator delete(void*) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==66850==    by 0x509305B: RDKit::Bond::~Bond() (Bond.cpp:42)
==66850==    by 0x4D55F7A: SmilesParseOps::CloseMolRings(RDKit::RWMol*, bool) (SmilesParseOps.cpp:472)
==66850==    by 0x4D378E0: RDKit::toMol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int (*)(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<RDKit::RWMol*, std::allocator<RDKit::RWMol*> >&), std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (SmilesParse.cpp:218)
==66850==    by 0x4D38811: RDKit::SmilesToMol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, RDKit::SmilesParserParams const&) (SmilesParse.cpp:363)
==66850==    by 0x1C1ED9: RDKit::SmilesToMol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, bool, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >*) (SmilesParse.h:79)
==66850==    by 0x13D4B2: testPass() (test.cpp:119)
==66850==    by 0x1BDA96: main (test.cpp:4321)
==66850==  Block was alloc'd at
==66850==    at 0x483BE63: operator new(unsigned long) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==66850==    by 0x50F1F17: RDKit::RWMol::createPartialBond(unsigned int, RDKit::Bond::BondType) (RWMol.cpp:444)
==66850==    by 0x4DB6484: yysmiles_parse(char const*, std::vector<RDKit::RWMol*, std::allocator<RDKit::RWMol*> >*, RDKit::Atom*&, RDKit::Bond*&, std::__cxx11::list<unsigned int, std::allocator<unsigned int> >*, void*, int&) (smiles.yy:215)
==66850==    by 0x4D36E33: RDKit::(anonymous namespace)::smiles_parse_helper(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<RDKit::RWMol*, std::allocator<RDKit::RWMol*> >&, RDKit::Atom*&, RDKit::Bond*&, int) (SmilesParse.cpp:112)
==66850==    by 0x4D37214: RDKit::(anonymous namespace)::smiles_parse(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<RDKit::RWMol*, std::allocator<RDKit::RWMol*> >&) (SmilesParse.cpp:149)
==66850==    by 0x4D3789A: RDKit::toMol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int (*)(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<RDKit::RWMol*, std::allocator<RDKit::RWMol*> >&), std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (SmilesParse.cpp:215)
==66850==    by 0x4D38811: RDKit::SmilesToMol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, RDKit::SmilesParserParams const&) (SmilesParse.cpp:363)
==66850==    by 0x1C1ED9: RDKit::SmilesToMol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, bool, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >*) (SmilesParse.h:79)
==66850==    by 0x13D4B2: testPass() (test.cpp:119)
==66850==    by 0x1BDA96: main (test.cpp:4321)

```

Reply:
I think this might have been checked in by accident.